src/init.js

```diff
@@ -41,6 +42,8 @@ function init(options) {
     var fileContent = file.contents.toString();
     var sourceMap;
 
+    file.newline = detectNewline.graceful(fileContent || '');
```

Comment:
https://github.com/gzzz/gulp-sourcemaps/commit/1ddace7b4f9ec431855729f11779631f5b9d993c

So, i've moved newline field into sourceMap object, but have found an issue with [gulp-uglify](https://github.com/terinjokes/gulp-uglify/).
If it is present in gulp pipe between sourcemaps init and write, [it recreates a sourceMap object](https://github.com/terinjokes/gulp-uglify/blob/master/minifier.js#L91) and drops newline field.

I think a ancillary file field – `sourcemapsNewline`, for example – is better way.